winpr/include/winpr/winpr.h

```diff
@@ -61,6 +61,20 @@
 #define WINPR_NORETURN(obj) obj
 #endif
 
+#ifdef BUILD_TESTING
+#define WINPR_LOCAL WINPR_API
+#else
+#if defined _WIN32 || defined __CYGWIN__
```

Comment:
Can't this just be
```
#if __GNUC__ >= 4
#define WINPR_LOCAL __attribute__((visibility("hidden")))
#else
#define WINPR_LOCAL
#endif
```
(no need to check for `WIN32` or `CYGWIN`)

Reply:
actually we only need this for windows as it is not as easy to export everything there. (with gcc/clang you just set global symbol visibility), but on windows you need the dll `__declspec` stuff.

Reply:
and second, `gcc` is also available on windows, where this does not apply (there you need the `dllexport` stuff with `gcc` too